lib/helper.ts

```diff
@@ -340,3 +340,14 @@ export function constructServiceUrl(
 
   return formattedUrl;
 }
+
+/**
+ * Returns true if and only if "mimeType" is a "JSON-like" mime type
+ * (e.g. "application/json; charset=utf-8").
+ * @param mimeType - the mimeType string
+ * @returns true if "mimeType" represents a JSON media type and false otherwise
+ */
+export function isJsonMimeType(mimeType: string) {
+  logger.debug(`Determining if the mime type '${mimeType}' specifies JSON content.`);
+  return !!mimeType && /^application\/json(;.*)?$/i.test(mimeType);
```

Comment:
small nit, I think it may also be valid for the `;` to be preceded by whitespace e.g.
The spec for [media type](https://www.rfc-editor.org/rfc/rfc9110.html#name-media-type) says:
>The type/subtype MAY be followed by semicolon-delimited parameters ([Section 5.6.6](https://www.rfc-editor.org/rfc/rfc9110.html#parameter)) in the form of name/value pairs.

In turn [5.6.6](https://www.rfc-editor.org/rfc/rfc9110.html#name-parameters) says:
>`parameters      = *( OWS ";" OWS [ parameter ] )`

where OWS is defined as optional white space by [5.6.3](https://www.rfc-editor.org/rfc/rfc9110.html#name-whitespace).

I can't say I've ever seen the first one preceded by whitespace, but it probably isn't worth tripping over.

Reply:
Good catch! I'll make that adjustment

Reply:
Regex updated and test added